packages/components/addon/components/hds/dialog/index.js

```diff
@@ -0,0 +1,101 @@
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { assert } from '@ember/debug';
+import { getElementId } from '../form/utils/getElementId';
+
+export const DEFAULT_SIZE = 'regular';
+export const DEFAULT_TYPE = 'neutral';
+export const SIZES = ['small', 'regular', 'large'];
+export const TYPES = ['neutral', 'warning', 'critical'];
+
+export default class HdsDialogIndexComponent extends Component {
+  @action
+  setupDialog(element) {
+    this.element = element;
+    element.showModal();
+  }
+
+  @action
+  onClick(event) {
+    // This may seem counterintuitive, but the <dialog> element is the target only when the backdrop is clicked
+    // otherwise the target would be a child of the <dialog>
+    if (event.target === this.element) {
+      this.closeDialog();
+    }
+  }
+
+  @action
+  closeDialog() {
+    // we call the "onClose" callback if it exists (and is a function)
+    if (this.args.onClose && typeof this.args.onClose === 'function') {
+      this.args.onClose();
+    }
+    this.element.close();
+  }
+
+  /**
+   * Sets the size of the dialog
+   * Accepted values: small, regular, large
+   *
+   * @param size
+   * @type {string}
+   * @default 'regular'
+   */
+  get size() {
+    let { size = DEFAULT_SIZE } = this.args;
+
+    assert(
+      `@size for "Hds::Dialog" must be one of the following: ${SIZES.join(
+        ', '
+      )}; received: ${size}`,
+      SIZES.includes(size)
+    );
+
+    return size;
+  }
+
+  /**
+   * Sets the type of the dialog
+   * Accepted values: neutral, warning, critical
+   *
+   * @param type
+   * @type {string}
+   * @default 'neutral'
+   */
+  get type() {
```

Comment:
not sold on the `@type` name, for multiple reasons:
- we have tried to avoid it as much as possible, to avoid problems/conflicts with the native `type` HTML attribute
- in the `Alert/Toast` component, this is called `@color` (see [here](https://design-system-components-hashicorp.vercel.app/components/alert#component-api)). I know @jorytindall doesn't like this, and I see his point, but I don't think we should have two different names for the same property on similar components

So I propose that we choose one of these options:

- we call this `@color` as in the `Alert/Toast` component, for consistency
- we rename the argument in the `Alert/Toast` component (refactoring the code) to whatever name we choose for this one ([`appearance`](https://atlassian.design/components/modal-dialog/examples)? [`intent`](https://evergreen.segment.com/components/dialog#default_with_a_danger_intent)? other?) and then refactor the code in the consumers (likely just using a simple codemod) to use the new argument name

I'm totally aware the second option has an intrinsic cost, but this would become technical debt in our codebase, and in my experience tech debt in a design system should be kept as low as possible, to avoid paying much higher "interests" later.

Reply:
@didoo I've considering raising this as an issue in the backlog to discuss as a team, maybe this is the time. I'm an advocate of **option 2** as well. We're getting our lines crossed a little bit with `@color` but not actually labelling the variant with a color (`warning`, `neutral`, etc).

Reply:
as discussed during team sync, I've update this to `@color`, for consistency and we can review these instances at a future point, across components.